spec/support/shared_examples_policy_machine.rb

```diff
@@ -820,4 +820,85 @@
 
   end
 
+
+  describe 'batch_find' do
+
+    before do
+      @one_fish = policy_machine.create_object('one:fish')
+      @two_fish = policy_machine.create_object('two:fish')
+      @red_one = policy_machine.create_object('red:one')
+      @read = policy_machine.create_operation('read')
+      @write = policy_machine.create_operation('write')
+      @u1 = policy_machine.create_user('u1')
+      @ua = policy_machine.create_user_attribute('ua')
+      [@one_fish, @two_fish, @red_one].each do |object|
+        policy_machine.add_association(@ua, Set.new([@read]), object)
+      end
+      @oa = policy_machine.create_object_attribute('oa')
+      policy_machine.add_association(@ua, Set.new([@write]), @oa)
+      policy_machine.add_assignment(@u1, @ua)
+      policy_machine.add_assignment(@red_one, @oa)
+    end
+
+    context 'when given a block' do
+
+      context 'and search terms' do
+        it 'returns the matching records' do
+          policy_machine.batch_find(type: :object, query: { unique_identifier: 'one:fish' }) do |batch|
+            expect(batch.size).to eq 1
+            expect(batch.first.unique_identifier).to eq 'one:fish'
+          end
+        end
+      end
+
+      context 'and config options' do
+        it 'returns the correct batch size' do
```

Comment:
Now that you're honoring batch size in the default implementation, mind passing in a batch size > 1 to document that?